### ISSUE TITLE
 Predis and phpredis compatible

### ISSUE THREAD

**Issue by CookieTime** [laravel/framework#20185]
- Laravel Version: 5.4
- PHP Version: 7.1
- Database Driver & Version:  redis 3.2

### Description:

when replacing redis driver from phpredis to predis,i got an error. 

### Steps To Reproduce:

```php
// phpredis 

Redis::set($lockName, $identifier, 'NX', $timeout, 'EX');

```

Change redis driver to phpredis . I got an error like this.

[
![error](https://user-images.githubusercontent.com/19209052/28405797-725232c0-6d60-11e7-9dfe-818fa0a27a46.jpg)
](url)

```php

// predis  
// this is correct 

Redis::set($lockName, $identifier, 'NX', 'EX', $timeout);

```

I recommend using this method without any modification.

**Comment by themsaid** [comment]
Should be:

```
Redis::set($lockName, $identifier, 'EX', $timeout, 'NX');
```

**Comment by CookieTime** [comment]
@themsaid 

> Redis::set($lockName, $identifier, 'EX', $timeout, 'NX');

when using phpredis driver like this,$lockName's ttl value will be forever.

**Comment by themsaid** [comment]
@CookieTime gotcha, thanks for the follow up. Will work on a solution :)

**Comment by lucasmichot** [comment]
@themsaid related failing build https://travis-ci.org/lucasmichot/framework/builds/25862338

**Comment by themsaid** [comment]
This is fixed in 5.5, can't fix for 5.4 since it might break people apps :/
